### PR TITLE
Add GRPC server query RPC endpoints

### PR DESCRIPTION
**Pull request by xichen2020**
cc @black-adder @cw9 @notbdu 

This PR adds gRPC server query RPC endpoints.

Will be addressed in followup PRs:
- [ ] gRPC client
- [ ] Integration tests

### CODE REVIEW

File: generated/proto/servicepb/query.proto [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+message OptionalCalculationOp {
+  oneof value {
+    bool no_value = 1;
+    Calculation.Op data = 2;
+  }
+}
+
+enum SortOrder {
+  UNKNOWNSORTORDER = 0;
```
Comment (cw9) [inline generated/proto/servicepb/query.proto]: Is this needed? Should it be by default `ASCENDING`? Or handle that logic in converting functions?

File: calculation/value.go [hunk 2]
```diff
@@ -45,6 +46,23 @@ func (u ValueUnion) MarshalJSON() ([]byte, error) {
 	}
 }
 
+// ToProto converts a value to a calculation value protobuf message.
+func (u ValueUnion) ToProto() servicepb.CalculationValue {
```
[the remark this answers is not in the document]
Comment (xichen2020) [reply calculation/value.go]: This is safe yeah. Sure can move panic to default.

File: generated/proto/servicepb/query.proto [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    ISNOTNULL = 14;
+    EXISTS = 15;
+    DOESNOTEXIST = 16;
+  }
+  string field = 1;
+  Op op = 2;
+  OptionalFilterValue value = 3 [(gogoproto.nullable) = false];
+}
+
+message OptionalString {
```
[the remark this answers is not in the document]
Comment (xichen2020) [reply generated/proto/servicepb/query.proto]: Sure np.

File: generated/proto/servicepb/query.proto [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  oneof value {
+    bool no_value = 1;
+    string data = 2;
```
[the remark this answers is not in the document]
Comment (xichen2020) [reply generated/proto/servicepb/query.proto]: This is only used in query proto to specify things like fields to filter against, so it's more readable to use strings on the client side.

File: generated/proto/servicepb/query.proto (hunk 1 above)
Comment (xichen2020) [reply generated/proto/servicepb/query.proto]: I generally prefer to have the default slot to be something unknown and make this explicitly optional if needed (i.e., handling in converting functions)

File: generated/proto/servicepb/service.proto [hunk 5]
```diff
@@ -5,49 +5,61 @@ option go_package = "github.com/xichen2020/eventdb/generated/proto/servicepb";
...
+  rpc QueryTimeBucket(TimeBucketQuery) returns (TimeBucketQueryResults) {}
 }
 
-message Field {
-  repeated string path = 1;
-  FieldValue value = 2 [(gogoproto.nullable) = false];
+message WriteResults {}
+
+message RawQueryResults {
+  repeated string raw = 1;
```
[the remark this answers is not in the document]
Comment (xichen2020) [reply generated/proto/servicepb/service.proto]: No this is to return a list of raw results, so you need to have an array here.

File: generated/proto/servicepb/value.proto [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  enum Type {
+    UNKNOWNTYPE = 0;
+    BOOL = 1;
+    NUMBER = 2;
+    STRING = 3;
+  }
+  Type type = 1;
+  bool bool_val = 2;
+  double number_val = 3;
+  string string_val = 4;
```
[the remark this answers is not in the document]
Comment (xichen2020) [reply generated/proto/servicepb/value.proto]: This is only used in the query filter in a query expression and is probably more readable to use strings. It should have no impact on performance.

**Pull request merged**